mycroft/skills/skill_loader.py

```diff
@@ -243,13 +244,52 @@ def _get_last_modified_time(path):
     # Ensure modification times are valid
     bad_times = _bad_mod_times(mod_times)
     if bad_times:
-        raise OSError('{} had bad modification times'.format(bad_times))
+        raise OSError(f'{bad_times} had bad modification times')
     if all_files:
         return max(os.path.getmtime(f) for f in all_files)
     else:
         return 0
 
 
+def get_skill_class(skill_module):
+    """Find MycroftSkill based class in skill module.
+
+    Arguments:
+        skill_module (module): module to search for Skill class
+
+    Returns:
+        (MycroftSkill): Found subclass of MycroftSkill or None.
+    """
+    candidates = []
+    for name, obj in skill_module.__dict__.items():
+        if isclass(obj):
+            if issubclass(obj, MycroftSkill) and obj is not MycroftSkill:
+                candidates.append(obj)
```

Comment:
i accounted for the base OVOS / Mycroft skill classes, but i don't have a good solution for derivative forks, i suppose ovos.conf can also be used for this ?

Reply:
i think that is out of scope for this PR, for those edge cases `create_skill` can be used as usual

Reply:
note that create_skill is given preference to auto detection of skill class, this is to allow some logic to be used in there to select a skill class

we have this planned in the setup skill to give a dedicated experience for voice/hybrid/gui experiences

Reply:
I think you addressed it; just need the subclass that isn't itself subclassed. I agree it shouldn't come up in legacy skills since they have `create_skill`, but mostly looking at how this could handle some arbitrary family of subclass skills (Some user or org who decides to move all their boilerplate into a subclass :wink: